File: apps/api/internal/provider/openrouter.go
package provider

import "github.com/manpreet/chatbot/apps/api/internal/config"

func NewOpenRouterClient(cfg config.ChatConfig) ChatCompleter {
	return newCompatibleClient(
		"openrouter",
		"openrouter",
		cfg.OpenRouterBaseURL,
		cfg.OpenRouterAPIKey,
		cfg.Model,
	)
}

File: apps/pyapi/pyapi/providers/openrouter.py
from __future__ import annotations

from pyapi.config import ChatConfig

from .compatible import OpenAICompatibleProvider


class OpenRouterProvider(OpenAICompatibleProvider):
    def __init__(self, config: ChatConfig):
        super().__init__(
            provider="openrouter",
            base_url=config.openrouter_base_url,
            api_key=config.openrouter_api_key,
            model=config.model,
        )
